src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -597,25 +597,29 @@ public void replacePlaceholders(ProgrammingExercise programmingExercise, Reposit
         }
-        else if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.SWIFT && programmingExercise.getProjectType() == ProjectType.PLAIN) {
-            fileService.replaceVariablesInDirectoryName(repository.getLocalPath().toAbsolutePath().toString(), "${packageNameFolder}", programmingExercise.getPackageName());
-            fileService.replaceVariablesInFileName(repository.getLocalPath().toAbsolutePath().toString(), "${packageNameFile}", programmingExercise.getPackageName());
-        }
-        else if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.SWIFT && programmingExercise.getProjectType() == ProjectType.XCODE) {
-            fileService.replaceVariablesInDirectoryName(repository.getLocalPath().toAbsolutePath().toString(), "${appName}", programmingExercise.getPackageName());
-            fileService.replaceVariablesInFileName(repository.getLocalPath().toAbsolutePath().toString(), "${appName}", programmingExercise.getPackageName());
+        else if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.SWIFT) {
+            if (programmingExercise.getProjectType() == ProjectType.PLAIN) {
+                fileService.replaceVariablesInDirectoryName(repository.getLocalPath().toAbsolutePath().toString(), "${packageNameFolder}", programmingExercise.getPackageName());
+                fileService.replaceVariablesInFileName(repository.getLocalPath().toAbsolutePath().toString(), "${packageNameFile}", programmingExercise.getPackageName());
+            }
+            if (programmingExercise.getProjectType() == ProjectType.XCODE) {
+                fileService.replaceVariablesInDirectoryName(repository.getLocalPath().toAbsolutePath().toString(), "${appName}", programmingExercise.getPackageName());
+                fileService.replaceVariablesInFileName(repository.getLocalPath().toAbsolutePath().toString(), "${appName}", programmingExercise.getPackageName());
+            }
         }
 
         Map<String, String> replacements = new HashMap<>();
 
         if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.JAVA || programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.KOTLIN) {
             replacements.put("${packageName}", programmingExercise.getPackageName());
         }
-        else if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.SWIFT && programmingExercise.getProjectType() == ProjectType.PLAIN) {
-            replacements.put("${packageName}", programmingExercise.getPackageName());
-        }
-        else if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.SWIFT && programmingExercise.getProjectType() == ProjectType.XCODE) {
-            replacements.put("${appName}", programmingExercise.getPackageName());
+        else if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.SWIFT) {
```

Comment:
I have left you a message on Slack providing an example of how this code could be replaced by using switch cases and extracting the programmingLanguage and ProjectType to variables beforehand, therefore the code should become way slimmer

Reply:
Thanks. I refactored it.